src/caduceus/caduceus_test.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"errors"
+	"github.com/Comcast/webpa-common/health"
+	"github.com/Comcast/webpa-common/logging"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+)
+
+// Begin mock declarations
+
+// mockHandler only needs to mock the `HandleRequest` method
+type mockHandler struct {
+	mock.Mock
+}
+
+func (m *mockHandler) HandleRequest(workerID int, inRequest CaduceusRequest) {
+	m.Called(workerID, inRequest)
+	return
+}
+
+// mockHealthTracker needs to mock things from both the `HealthTracker`
+// interface as well as the `health.Monitor` interface
+type mockHealthTracker struct {
+	mock.Mock
+}
+
+func (m *mockHealthTracker) SendEvent(healthFunc health.HealthFunc) {
+	m.Called(healthFunc)
+	return
+}
+
+func (m *mockHealthTracker) ServeHTTP(response http.ResponseWriter, request *http.Request) {
+	m.Called(response, request)
+	return
+}
+
+func (m *mockHealthTracker) IncrementBucket(inSize int) {
+	m.Called(inSize)
+	return
+}
+
+// Begin test functions
+
+func TestMain(m *testing.M) {
+	os.Exit(m.Run())
+}
+
+func TestWorkerPool(t *testing.T) {
+	assert := assert.New(t)
+
+	workerPool := WorkerPoolFactory{
+		NumWorkers: 1,
+		QueueSize:  1,
+	}.New()
+
+	t.Run("TestWorkerPoolCreation", func(t *testing.T) {
+		assert.NotNil(workerPool)
+	})
+
+	t.Run("TestWorkerPoolSend", func(t *testing.T) {
+		err := workerPool.Send(func(workerID int) {
+			// do nothing
+		})
+
+		assert.Nil(err)
+	})
+
+	workerPool = WorkerPoolFactory{
+		NumWorkers: 1,
+		QueueSize:  0,
+	}.New()
+
+	t.Run("TestWorkerPoolFullQueue", func(t *testing.T) {
+		err := workerPool.Send(func(workerID int) {
+			// do nothing
+		})
+
+		assert.NotNil(err)
+	})
+}
+
+func TestIncrementBucket(t *testing.T) {
+	assert := assert.New(t)
+
+	testData := []struct {
+		inSize       int
+		expectedStat health.Stat
+	}{
+		{inSize: 99, expectedStat: PayloadsOverZero},
+		{inSize: 999, expectedStat: PayloadsOverHundred},
+		{inSize: 9999, expectedStat: PayloadsOverThousand},
+		{inSize: 10001, expectedStat: PayloadsOverTenThousand},
+	}
+
+	t.Run("TestIncrementBucket", func(t *testing.T) {
+		for _, data := range testData {
+			fakeMonitor := new(mockHealthTracker)
+			fakeMonitor.On("SendEvent", mock.AnythingOfType("health.HealthFunc")).Run(
+				func(args mock.Arguments) {
+					healthFunc := args.Get(0).(health.HealthFunc)
+					stats := make(health.Stats)
+
+					healthFunc(stats)
+					assert.Equal(1, stats[data.expectedStat])
+				}).Once()
+
+			caduceusHealth := &CaduceusHealth{fakeMonitor}
+
+			caduceusHealth.IncrementBucket(data.inSize)
+			fakeMonitor.AssertExpectations(t)
+		}
+	})
+}
+
+func TestServeHTTP(t *testing.T) {
+	assert := assert.New(t)
```

Comment:
Test names should refer to a type first, not a method.  TestFooServeHTTP would be more consistent with golang style, where Foo was a type that had a ServeHTTP method.  Alternatively, call this TestFoo and use the new t.Run(...) syntax for each method or scenario of Foo.